cirq-core/cirq/transformers/stratify.py

```diff
@@ -73,58 +93,87 @@ def stratified_circuit(
     reversed_circuit = circuit[::-1]
     solutions = []
     for c in classifiers_permutations:
-        solutions.append(stratify_circuit(list(c), circuit))
+        solutions.append(
+            stratify_circuit(
+                circuit,
+                classifiers=list(c),
+                context=context or transformer_api.TransformerContext(),
+            )
+        )
         # Do the same thing, except this time in reverse. This helps for some
         # circuits because it inserts operations at the end instead of at the
         # beginning.
-        solutions.append(stratify_circuit(list(c), reversed_circuit)[::-1])
+        solutions.append(
+            stratify_circuit(
+                reversed_circuit,
+                classifiers=list(c),
+                context=context or transformer_api.TransformerContext(),
+            )[::-1]
+        )
 
     # Return the shortest circuit.
     return min(solutions, key=lambda c: len(c))
 
 
-def stratify_circuit(classifiers: Iterable[Classifier], circuit: circuits.Circuit):
+def stratify_circuit(
```

Comment:
Should we make this a private method with a leading underscore just to make clear we don't expect people to use it ?

Reply:
done